### PR TITLE
Fix memory leak when set observers in MapboxLogger and MapboxMetricsReporter

### PR DESCRIPTION
**Pull request by Lebedinsky**
## Description

Fix memory leak when set observers in `MapboxLogger` and `MapboxMetricsReporter` [issue comment](https://github.com/mapbox/mapbox-navigation-android/pull/2330#pullrequestreview-336183666)

- [ ] I have added any issue links
- [ ] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [ ] I have added the appropriate milestone and project boards

### Goal

When we add observers (for example some Activity) to `MapboxLogger` and `MapboxMetricsReporter` and then we destroy this activity (leave them), memory leak happens. This PR is about fix this issue.

### Implementation

Added removeObserver() methods in `MapboxLogger` and `MapboxMetricsReporter` and call them in `onDestroy()` method of activity

## Testing

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

**Comment by korshaknn** [comment]
would it be better to use `LifecycleOwner` and `LifecycleObserver`? 
so 'Logger' and `MetricsReporter` will observe and stop when need.
wdyt?

**Comment by Lebedinsky** [comment]
@korshaknn Yeah, it make sense, but what about case if we need to add this observer not in activity, but in some other class which not inherited from `LifecycleOwner`? I think we shouldn't make it depends on Lifecycle... wdyt?

**Comment by RingerJK** [comment]
not sure that it will work as expected, user doesn't always want that `MetricsReporter` depends on `LifecycleOwner`

File: libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt [hunk 1]
```diff
@@ -70,6 +70,7 @@ object MapboxMetricsReporter : MetricsReporter {
      */
     @JvmStatic
     fun disable() {
+        removeObserver()
```
Comment (RingerJK) [inline libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt]: hm, `removeObserver` should be independent on  enable-disable 🤔make sense?

Comment (Lebedinsky) [reply libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt]: Yeah, that's why we have separate method removeObserver()... But sometimes disable() calls in some places where user have no control (service for example)... so this way is more safety. Wdyt?

Comment (LukasPaczos) [reply libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt]: It seems like there's no way to re-enable a metrics session (should there be?), so it makes sense to clean up all the listeners as well.

File: libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt [hunk 2]
```diff
@@ -87,4 +88,8 @@ object MapboxMetricsReporter : MetricsReporter {
     override fun setMetricsObserver(metricsObserver: MetricsObserver) {
         this.metricsObserver = metricsObserver
     }
+
+    fun removeObserver() {
```
Comment (Guardiola31337) [inline libnavigation-metrics/src/main/java/com/mapbox/navigation/metrics/MapboxMetricsReporter.kt]: Should be this part of the `MetricsReporter` `interface` / contract?

**Comment by Guardiola31337** [comment]
Unrelated to the changes introduced as part of this PR but I run into the following leak when testing 👀 

offboard_router_activity_toast_leak

**Pull request merged**